websites.qmd

```diff
@@ -1 +1,1752 @@
+One approach to automating publishing is to check your website source into version control and let a provider handle the automation.
+From your perspective, you run `quarto render` locally, then push the changes to your repository. 
+Your provider then updates the deployed site.
+
+This approach is the easiest to set up---you generally don't need to do anything except connect your repo to your CI service. 
+However, you can end up with commits with large diffs (the changes to the source documents, and all the changes to the rendered output in `_site/` or `docs/`). 
+These large diffs are cumbersome to review, and take longer to send to your Git remote host.
+
+Two examples of this approach are:
+
+-  Github deploy from `docs/` (as described in @sec-publishing-github). GitHub is the host for your source code, the provider of the CI and the publishing venue.
+
+-  Netlify deploy from a Git provider (as described in @sec-publishing-netlify). Any supported Git provider is the host for your source code, and Netlify is the CI provider and publishing venue.
+
+Alternatively, you could set up your own automation (e.g. via [GitHub Actions](#github-actions)) that deploys your content on any push to your repo. 
+
+### Local code execution and remote render
+
+Local code execution, with remote rendering provides a good middle ground.
+You aren't burdened with setting up, and maintaining a remote code execution environment,
+but you also don't have to store a whole lot of rendered output in your repo.
+
+This approach requires using `freeze` (@sec-freeze-cache) to store the results of the code execution.
+You run `quarto render` locally, as required, to generate `_freeze/`.
+You then check `_freeze/`, but not `_site/`, into version control.
+Your CI agent then uses the Quarto CLI to render and publish your site.
+
+The most common CI service for this approach is GitHub. 
+The project source is stored in a GitHub repo, and a [GitHub Action](#github-actions) is used to setup Quarto, 
+and run the required Quarto commands.
+The Quarto command could be as simple as `quarto publish {venue}`, which allows your publishing venue to be any of those supported by `quarto publish`.
+
+
+### Remote code execution and remote render
+
+Remote code execution combined with remote rendering provides the most protection for keeping your work reproducible---your CI will fail as soon as it isn't.
+However, it comes with the most setup, maintenance, and cost (time and potentially money) since all the work is happening on a remote server.
+
+You now only check your source documents (e.g. `.qmd` files) into version control. 
+Your CI service needs to install your computational environment, so you likely will also create and check-in a record of the packages required (e.g. using `renv` for R, or `requirements.txt`, `pipenv`, or `uv` for Python).  
+When your CI runs, it will install your computational environment, install Quarto, and then run a full render and publish.  
```

Comment:
Wonder why this sentence is on the next line in the rendered book?

Reply:
Fixed.